### ISSUE TITLE
Ragavi plotting woes

### ISSUE THREAD

**Issue by KshitijT** [caracal-pipeline/caracal#1041]
When the plotter is set to 'ragavi_vis', I get the following error:
```
Traceback (most recent call last):
  File "/usr/local/bin/ragavi-vis", line 11, in <module>
    main(options=options)
  File "/usr/local/lib/python3.6/dist-packages/ragavi/visibilities.py", line 1440, in main
    colour_axis=colour_axis)
  File "/usr/local/lib/python3.6/dist-packages/ragavi/visibilities.py", line 1169, in hv_plotter
    ms_name=ms_name, xds_table_obj=xds_table_obj)
  File "/usr/local/lib/python3.6/dist-packages/ragavi/visibilities.py", line 906, in gen_grid
    s_agg = agg.sel(**{cat: c_val})
  File "/usr/local/lib/python3.6/dist-packages/xarray/core/dataarray.py", line 1061, in sel
    **indexers_kwargs,
  File "/usr/local/lib/python3.6/dist-packages/xarray/core/dataset.py", line 2066, in sel
    self, indexers=indexers, method=method, tolerance=tolerance
  File "/usr/local/lib/python3.6/dist-packages/xarray/core/coordinates.py", line 397, in remap_label_indexers
    obj, v_indexers, method=method, tolerance=tolerance
  File "/usr/local/lib/python3.6/dist-packages/xarray/core/indexing.py", line 270, in remap_label_indexers
    idxr, new_idx = convert_label_indexer(index, label, dim, method, tolerance)
  File "/usr/local/lib/python3.6/dist-packages/xarray/core/indexing.py", line 190, in convert_label_indexer
    label.item(), method=method, tolerance=tolerance
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/indexes/base.py", line 2648, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 998, in pandas._libs.hashtable.Int64HashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1005, in pandas._libs.hashtable.Int64HashTable.get_item
KeyError: 18
Traceback (most recent call last):
  File "/stimela_mount/code/run.py", line 38, in <module>
    subprocess.check_call(shlex.split(_runc))
  File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ragavi-vis', '--ms', '/stimela_mount/msdir/1565287561_sdp_l0.ms', '--xaxis', 'real', '--yaxis', 'imaginary', '--canvas-height', '300', '--canvas-width', '300', '--cmap', 'blues', '--corr', '0', '--data-column', 'CORRECTED_DATA', '--field', '0', '--htmlname', '/stimela_mount/output/mypipelinerun-1565287561_sdp_l0-1gc-J1939-6342-real_imag-bpcal-_Corr_XX', '--iter-axis', 'scan', '--mem-limit', '1GB']' returned non-zero exit status 1
singularity returns error code 1
job failed at 2020-05-03 01:34:47.818132 after 0:09:07.729361
Traceback (most recent call last):
  File "/scratch/users/kshitij/Stimela/stimela/recipe.py", line 577, in run
                                                                                                                            45
```

The options are given by the lines (as above, so below):
```
'/stimela_mount/msdir/1565287561_sdp_l0.ms', '--xaxis', 'real', '--yaxis', 'imaginary', '--canvas-height', '300', '--canvas-width', '300', '--cmap', 'blues', '--corr', '0', '--data-column', 'CORRECTED_DATA', '--field', '0', '--htmlname', '/stimela_mount/output/mypipelinerun-1565287561_sdp_l0-1gc-J1939-6342-real_imag-bpcal-_Corr_XX', '--iter-axis', 'scan', '--mem-limit', '1GB']' returned non-zero exit status 1

```

Just as a curiosity, is 1GB memory enough for this? Putting at defcon-0, lower if @o-smirnov and @Mulan-94 think it is not that dire.

**Comment by gigjozsa** [comment]
@o-smirnov Do we close this or do we only remove the milestone tag?

**Comment by o-smirnov** [comment]
Woebegone.

**Comment by gigjozsa** [comment]
Not really nice to reopen this again, but on my laptop running since 20 mins:
```
2020-05-12 10:12:03 CARACal.Stimela.plotgains-B-0-0 INFO: job started at 2020-05-12 10:12:03.093540
# 4f4125360079e6259ceac32e134a9dab3e9f6c70250ba246a7f6d9557aba6227
# 12.05.2020@08:12:06 - ragavi       - INFO       - Acquiring table: mypipelinerun-1532022061_subset-1gc1_primary.B0
# 12.05.2020@08:12:06 - ragavi       - INFO       - Spw: 0, Field: J1331+3030, Corr: 0 amplitude
# 12.05.2020@08:12:10 - ragavi       - INFO       - Spw: 0, Field: J1331+3030, Corr: 1 amplitude
```

At this stage we should at least put out a warning...

**Comment by gigjozsa** [comment]
```
Stimela release: 1.4.6
Stimela build: a0af56a43950dedf949ebce7ff2a00fa8a278ef4
f091b75b459c2cb31d4c6b057e9f68db46562fb5
from: https://github.com/ratt-ru/Stimela
```

**Comment by gigjozsa** [comment]
I see that the release version is wrong.

**Comment by gigjozsa** [comment]
@SpheMakh could you update the release number in Stimela/setup.py ?

**Comment by gigjozsa** [comment]
This is slightly urgent I guess...
I issued a PR on Stimela, which might fix this: https://github.com/ratt-ru/Stimela/pull/612

**Comment by o-smirnov** [comment]
It is urgent, but another PR has to go in for a true fix. Give it an hour.

**Comment by gigjozsa** [comment]
@o-smirnov @SpheMakh ok let me know when I can test again.

**Comment by paoloserra** [comment]
> Not really nice to reopen this again, but on my laptop running since 20 mins

Docker or Singularity? I've had no problems with ragavi-gains on Docker since Caracal release, but maybe I've been lucky so far?

**Comment by edeblok** [comment]
ragavi-gains has been working well for me since yesterday as well on two different measument sets (docker version).

**Comment by o-smirnov** [comment]
Well the fix we figured out yesterday actually didn't make it into a Stimela release until now... ooops... so you have been a couple of very lucky boys.

But yes, now that Stimela 1.5.1 is out, another round of hard testing will be good (running for me fine so far...)

**Comment by SpheMakh** [comment]
Fixed in stimela 1.5.1
